java/com/google/flatbuffers/Table.java

```diff
@@ -239,6 +243,28 @@ protected static int compareStrings(int offset_1, int offset_2, ByteBuffer bb) {
     if (len_1 > len_2) return 1;
     return 0;
   }
+  
+  /**
+   * Compare string from the buffer with the 'String' object.
+   *
+   * @param offset_1 An 'int' index of the first string into the bb.
+   * @param key Second string.
+   * @param bb A {@code ByteBuffer} to get the first string.
+   */
+  protected static int compareStrings(int offset_1, String key, ByteBuffer bb) {
+    offset_1 += bb.getInt(offset_1);
+    int len_1 = bb.getInt(offset_1);
+    int len_2 = key.length();
+    int startPos_1 = offset_1 + Constants.SIZEOF_INT;
+    int len = Math.min(len_1, len_2);
+    for (int i = 0; i < len; i++) {
+      if (bb.array()[i + startPos_1] != key.charAt(i))
```

Comment:
We should compare only a segment of the `bb.array()` , so I can get it before the loop in `lookupByKey`, pass it to `compareStrings` as an argument and then use `Arrays.equals`. Either I can get this segment  in `compareStrings`, but I don't think it is a good idea. Or maybe would be better to leave the comparison without `Arrays.equals`?